site/app/models/gradeable/GradedGradeable.php

```diff
@@ -77,6 +77,22 @@ public function getGradeable() {
         return $this->gradeable;
     }
 
+    /**
+     * Gets if the submitter submitted on time
+     * @return bool
+     */
+    public function isOnTimeSubmission(User $user) {
+        if ($this->gradeable->isTeamAssignment()) {
+            $user = $this->getSubmitter()->getTeam()->getMemberUsersSorted()[0];
+        }
+        else {
+            $user = $this->getSubmitter()->getUser();
+        }
+        $late_day = new LateDays($this->gradeable->core, $user, [$this]);
```

Comment:
Probably don't want to create a new LateDay here,
`$late_days = LateDays::fromUser($this->core, $this->core->getUser());`

would be a little better but still not idea since it needs to make a database query per student